### PR TITLE
Added password length check

### LINKED ISSUE #430: "Unknown" Error on login - Looks like password is too long
## Describe the bug

When trying to add integration after entering User/Pass an "Unknown error occurred" error message appears. Looking at the HA Core logs the error in the logs section below appears, suggesting that the 100 character password I generated is being rejected by the cypher. Shortening the password bypassed the issue.

## Version of the integration

If you are unsure about the version check the [github release here](https://github.com/leikoilja/ha-google-home/releases).

```text
1.9.4
```

## Logs

```
2022-01-09 17:08:16 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 220, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 78, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 181, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 137, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 157, in post
    return await super().post(request, flow_id)
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 62, in wrapper
    result = await method(view, request, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 252, in async_configure
    result = await self._async_handle_step(flow, cur_step["step_id"], user_input)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 325, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/config/custom_components/google_home/config_flow.py", line 61, in async_step_user
    master_token = await self._test_credentials(client)
  File "/config/custom_components/google_home/config_flow.py", line 96, in _test_credentials
    master_token = await client.async_get_master_token()
  File "/config/custom_components/google_home/api.py", line 75, in async_get_master_token
    master_token = await self.hass.async_add_executor_job(_get_master_token)
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/google_home/api.py", line 73, in _get_master_token
    return self._client.get_master_token()
  File "/usr/local/lib/python3.9/site-packages/glocaltokens/client.py", line 218, in get_master_token
    res = perform_master_login(
  File "/usr/local/lib/python3.9/site-packages/gpsoauth/__init__.py", line 127, in perform_master_login
    "EncryptedPasswd": google.construct_signature(
  File "/usr/local/lib/python3.9/site-packages/gpsoauth/google.py", line 58, in construct_signature
    encrypted_login = cipher.encrypt((email + "\x00" + password).encode("utf-8"))
  File "/usr/local/lib/python3.9/site-packages/Cryptodome/Cipher/PKCS1_OAEP.py", line 115, in encrypt
    raise ValueError("Plaintext is too long.")
ValueError: Plaintext is too long.
```

### PR DESCRIPTION
**Pull request by ArnyminerZ**
Should fix #430, not yet tested.

### CODE REVIEW

**Review by leikoilja — COMMENTED** [review]
This is awesome!
However, i would argue that we should move this validation to `glocaltokens` package, since it is it's job to take in input parameters and return auth tokens. It is also probably `glocaltoken` that is freezing out if too long password is given. 
We can move this validation to `glocaltokens` and raise an error during authentication. Here, we can capture the error and show the UI error message. That approach will also allow us to unittest lengthy passwords in `glocaltokens`. What do you think?

**Comment by ArnyminerZ** [comment]
@leikoilja I've opened leikoilja/glocaltokens#229 for this

File: custom_components/google_home/config_flow.py [hunk 1]
```diff
@@ -51,22 +51,27 @@ async def async_step_user(
...
-                config_data[CONF_USERNAME] = user_input[CONF_USERNAME]
-                config_data[CONF_PASSWORD] = user_input[CONF_PASSWORD]
-                config_data[CONF_MASTER_TOKEN] = master_token
-                config_data[CONF_ANDROID_ID] = await client.get_android_id()
-                return self.async_create_entry(title=username, data=config_data)
-            self._errors["base"] = "auth"
+
+            if len(password) >= 100:
+                self._errors["base"] = "pass-len"
+            else:
```
Comment (leikoilja) [inline custom_components/google_home/config_flow.py:59]: Just a matter of coding preference but it's nice to have a first block of `if` being "the longest". 
Like:
```python
from .const import MAX_PASSWORD_LENGTH
...
if len(password) < MAX_PASSWORD_LENGTH:
  ... # all the major logic
else:
  self._errors["base"] = "pass-len"
```
but it's purely up to the way you like it :) 
However, please move out the static `100` into a `const` for easier re-usability and style :)

**Review by leikoilja — CHANGES_REQUESTED** [review]

**Review by leikoilja — APPROVED** [review]
This is awesome! Thanks @ArnyminerZ for the implementation 💥 
Can you please make sure the linting checks passes and merge in when happy with it?

**Pull request merged**